components/tikv_util/src/mpsc/batch.rs

```diff
@@ -241,16 +241,25 @@ impl<T> Stream for Receiver<T> {
     }
 }
 
+/// A Collector Used in `BatchReceiver`.
+pub trait BatchCollector<Collection, Elem> {
+    /// If `elem` is collected into `collection` successfully, return `None`.
+    /// Otherwise return `elem` back.
+    fn collect(&mut self, collection: &mut Collection, elem: Elem) -> Option<Elem>;
+    /// `reset` will be called after a batch is spilled out.
+    fn reset(&mut self) {}
```

Comment:
Is reset necessary? How about assuming batch will be spilled out when collect return `Some`? So that collector can reset itself in `collect`.